job_dsl/rpc_gating_merge_trigger.groovy

```diff
@@ -16,58 +16,53 @@ common.globalWraps(){
     return
   }
 
-  stage("Build-Gating-Venv") {
-    build(
-      job: "Build-Gating-Venv",
-      wait: true,
-      parameters: [
-        [
-          $class: 'StringParameterValue',
-          name: 'RPC_GATING_BRANCH',
-          value: env.RPC_GATING_BRANCH
-        ]
-      ]
-    ) // build
-  } // stage
+  // Here we define the jobs to execute when a merge happens.
+  // Some are executed in series, and some parallel. We wait
+  // until they are all complete before completing the pipeline.
 
-  stage("Jenkins-Job-Builder") {
-    build(
-      job: "Jenkins-Job-Builder",
-      wait: true,
-      parameters: [
-        [
-          $class: 'StringParameterValue',
-          name: 'RPC_GATING_BRANCH',
-          value: env.RPC_GATING_BRANCH
-        ]
-      ]
-    ) // build
-  } // stage
+  List serialJobs = [
+    'Build-Gating-Venv',
+    'Jenkins-Job-Builder'
+  ]
 
-  // To prevent having to continually update a list of jobs to run
-  // in parallel once the two blocking jobs are complete, we match
-  // the jobs against a regex and execute everything that matches
-  // it in parallel.
-  jobNames = common.getMatchingJobs("(?:Build-Summary-Docker-Build|Setup-.*)")
+  List parallelJobs = [
+    'Build-Summary-Docker-Build',
+    'Setup-Grafana',
+    'Setup-Graphite',
+    'Setup-Nodepool',
+    'Setup-StatsD'
+  ]
+
+  List jobParameters = [
+    [
+      $class: 'StringParameterValue',
+      name: 'RPC_GATING_BRANCH',
+      value: env.RPC_GATING_BRANCH
+    ]
+  ]
+
+  for (job in serialJobs) {
+    stage(job) {
+      build(
+        job: job,
+        wait: true,
+        parameters: jobParameters
+      ) // build
+    } // stage
+  } // for
 
   def parallelBuilds = [:]
 
   // Cannot do for (job in jobNames), see:
   // https://jenkins.io/doc/pipeline/examples/#parallel-multiple-nodes
-  for (j in jobNames) {
+  for (j in parallelJobs) {
     def job = j
     parallelBuilds[job] = {
-      stage(job) {
+      job: {
```

Comment:
This syntax doesn't look correct. I think you want to remove this line. It looks like you're trying to define a `Map` with this line but it's happening inside a `Closure`. I'm pretty sure `parallelBuilds[job]` is already doing what this line wants.

Reply:
Ah, silly me - I think you're right. Good catch!